### PR TITLE
Add modslist formspec for /mods command.

### LINKED ISSUE #9563: Formspec for /mods
## Problem

The `/mods` command now prints a list of installed mods into chat, which can get very messy the more mods are installed.

## Solutions

As it has been done with `/help` I'd suggest to also list the installed mods in a formspec, for better readabitily.

### LINKED ISSUE #10367: Show `/mods` Command result in a formspec
## Problem

 This is what you can see if you server have a lot of mods:
screenshot_20200905_120511
This will make my client lag and make i cant see anything.

## Solutions

Put mods info in a formspec, that include mod technical name, `name` in `mod.conf`, and description.

Also put mods in group, group can be modpack.

### PR DESCRIPTION
**Pull request by Andrey2470T**
I've removed the branch of #10450 and closed that PR because some irrelevant changes got caught to there from others` PRs.

-----------------------------------------------------------------------

Fixes #10367 and fixes #9563 .
Screenshot_20201001_165954

## To do

This PR is Ready for Review.

## How to test
1. Join a world.
2. Call /mods command.

### CODE REVIEW

File: builtin/common/information_formspecs.lua [hunk 1]
```diff
@@ -20,6 +20,14 @@ local LIST_FORMSPEC_DESCRIPTION = [[
 		button_exit[5,7;3,1;quit;%s]
 	]]
 
+local LIST_FORMSPEC_MODSLIST = [[
+		size[6,8]
+		label[0,-0.1;%s]
+		tablecolumns[color;tree;text]
+		table[0,0.5;6,7;modslist;%s;0]
```
Comment (hlqkj) [inline builtin/common/information_formspecs.lua:27]: Wouldn't be better to use a `5.8` width? It's more consistent with the other formspecs.

Comment (Andrey2470T) [reply builtin/common/information_formspecs.lua:27]: Do you mean the width of the formspec window itself or the width of the table?

Comment (hlqkj) [reply builtin/common/information_formspecs.lua:27]: Sorry, meant the table width

Comment (Andrey2470T) [reply builtin/common/information_formspecs.lua:27]: I am not sure about the consistency with other formspecs, but in any case it looks like consistent with its formspec window and well-centered on that.

Comment (hlqkj) [reply builtin/common/information_formspecs.lua:27]: Yup it is, if you look at [#L9](https://github.com/minetest/minetest/blob/8a50fd2f9e711d06de7f55ba0e9015622535b217/builtin/common/information_formspecs.lua#L9) and [#L17](https://github.com/minetest/minetest/blob/8a50fd2f9e711d06de7f55ba0e9015622535b217/builtin/common/information_formspecs.lua#L17) you'll notice they used the same `.8` width, which for the relevant formspecs sizes gives the same right margin : )

File: builtin/game/chat.lua [hunk 2]
```diff
@@ -666,7 +666,9 @@ core.register_chatcommand("mods", {
 	description = S("List mods installed on the server"),
 	privs = {},
 	func = function(name, param)
-		return true, table.concat(core.get_modnames(), ", ")
+		core.show_formspec(name, "modslist",
+			build_mods_formspec())
```
Comment (appgurueu) [inline builtin/game/chat.lua:670]: Mod formspec might be cached on start.

Comment (Andrey2470T) [reply builtin/game/chat.lua:670]: How would do that? Just declare a new varyable with the cache at the beginning of the file?

Comment (appgurueu) [reply builtin/game/chat.lua:670]: > How would do that? Just declare a new varyable with the cache at the beginning of the file?

Yes, like that.

**Comment by Andrey2470T** [comment]
Hmm, why was it closed allegedly from my name when I had force-pushed the changes to the branch? I didn't close it :\ How to open it now?

**Comment by sfan5** [comment]
Under certain conditions force-pushing a branch can close the PR and render it unusable. You'll have to open a new one with different branch.

**Pull request closed without merging**